checker.go

```diff
@@ -482,6 +482,14 @@ func (c *notChecker) Check(got interface{}, args []interface{}, note func(key st
 	return errors.New("unexpected success")
 }
 
+// IsNonNil returns a Checker negating the IsNil checker. IsNonNil
```

Comment:
I think I'd go with `IsNotNil` here (or maybe just `NotNil` which is what the original [gocheck](https://github.com/go-check/check/blob/v1/checkers.go#L152) checker was named).

Reply:
+1 for `IsNotNil`

Reply:
New commit added to go back to `IsNotNil` - one question though - "please include the docs in the package-level docs along with all the other checks" - not sure what you mean by this? Sorry - should I also put the comment about this function somewhere else?

Reply:
I didn't touch the README as I thought it was generated, but I'm happy to add as well if that's what you mean? I was talking to @frankban about helping with the README if you decide to go non-generated.